airflow/providers/teradata/hooks/teradata.py

```diff
@@ -0,0 +1,212 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""A Airflow Hook for interacting with Teradata SQL Server."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, TypeVar
+
+import sqlalchemy
+import teradatasql
+from teradatasql import TeradataConnection
+
+from airflow.providers.common.sql.hooks.sql import DbApiHook
+
+T = TypeVar("T")
+if TYPE_CHECKING:
+    from airflow.models.connection import Connection
+
+
+class TeradataHook(DbApiHook):
+    """General hook for interacting with Teradata SQL Database.
+
+    This module contains basic APIs to connect to and interact with Teradata SQL Database. It uses teradatasql
+    client internally as a database driver for connecting to Teradata database. The config parameters like
+    Teradata DB Server URL, username, password and database name are fetched from the predefined connection
+    config connection_id. It raises an airflow error if the given connection id doesn't exist.
+
+    See :doc:` docs/apache-airflow-providers-teradata/connections/teradata.rst` for full documentation.
```

Comment:
This won't render in API documentation.
You could check documentation locally by run:

```console
breeze build-docs teradata
```

Instead of reference to the rst files, better use targets, because it do not breaks if page moved into the future.


Example
---

**Define target**
https://github.com/apache/airflow/blob/faa32f23e824ec8dd00b296ce9d8bd239ac0437f/docs/apache-airflow-providers-slack/connections/slack.rst?plain=1#L20-L23

You've already defined it into the `docs/apache-airflow-providers-teradata/connections/teradata.rst`

**Use it into the docstrings**
https://github.com/apache/airflow/blob/46470aba68e5ebeee24a03dc22d012a50ee287ad/airflow/providers/slack/hooks/slack.py#L67

https://github.com/apache/airflow/blob/46470aba68e5ebeee24a03dc22d012a50ee287ad/airflow/providers/slack/hooks/slack.py#L92-L93

Reply:
Updated to use targets instead of rst files. Ran documentation in local as https://github.com/Teradata/airflow/actions/workflows/ci-teradata-documentation.yml. It looks good now.